### PR TITLE
Improve jsDocs

### PR DESCRIPTION
**Pull request by WiXSL**

### CODE REVIEW

File: packages/ra-core/src/actions/listActions.ts [hunk 1]
```diff
@@ -71,7 +71,7 @@ export interface ToggleListItemExpandAction {
  * Action creator to toggle the expanded state of a list item
  *
  * @param {string} resource The resource name, e.g. 'posts'
- * @param {string|integer} id The record identifier, e.g. 123
+ * @param {string|int} id The record identifier, e.g. 123
```
Comment (fzaninotto) [inline packages/ra-core/src/actions/listActions.ts:74]: Where do you see that `int` exists? I don't see it in https://jsdoc.app/tags-param.html#names-types-and-descriptions or https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#type

If anything, the `integer` should be changed into the native JavaScript type `Number`, but it carries a lesser constraint. So I'm -1 for this change, in all the files you've edited.

File: packages/ra-core/src/controller/field/useReferenceArrayFieldController.ts [hunk 2]
```diff
@@ -42,12 +42,14 @@ const defaultSort = { field: null, order: null };
  *      source: 'referenceIds';
  * });
  *
- * @param {Object} option
```
Comment (fzaninotto) [inline packages/ra-core/src/controller/field/useReferenceArrayFieldController.ts:45]: The former syntax is also valid, see https://jsdoc.app/tags-param.html#parameters-with-properties and https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#typedef-callback-and-param

So your change isn't required nor necessary - just a matter of preference. I'd rather not change these types.

File: packages/ra-ui-materialui/src/button/DeleteButton.tsx [hunk 3]
```diff
@@ -11,7 +11,7 @@ import DeleteWithConfirmButton from './DeleteWithConfirmButton';
  * Button used to delete a single record. Added by default by the <Toolbar> of edit and show views.
  *
  * @typedef {Object} Props The props you can use (other props are injected if you used it in the <Toolbar>)
- * @param {Prop} props
+ * @param {Props} props
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/button/DeleteButton.tsx:14]: If you edit this one, you'd better move it out of the `typedef` definition - I don't think `@param` is valid under `@typedef`

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: packages/ra-core/src/controller/field/useReferenceArrayFieldController.ts (hunk 2 above)
Comment (WiXSL) [reply packages/ra-core/src/controller/field/useReferenceArrayFieldController.ts:45]: Done!

File: packages/ra-core/src/actions/listActions.ts (hunk 1 above)
Comment (WiXSL) [reply packages/ra-core/src/actions/listActions.ts:74]: Done!

File: packages/ra-ui-materialui/src/button/DeleteButton.tsx (hunk 3 above)
Comment (WiXSL) [reply packages/ra-ui-materialui/src/button/DeleteButton.tsx:14]: Done!

File: packages/ra-core/src/controller/field/useReferenceArrayFieldController.ts (hunk 2 above)
Comment (fzaninotto) [reply packages/ra-core/src/controller/field/useReferenceArrayFieldController.ts:45]: the change is still present

File: packages/ra-core/src/controller/input/useReferenceArrayInputController.ts [hunk 4]
```diff
@@ -28,12 +28,14 @@ import { useResourceContext } from '../../core';
  *      source: 'referenceIds';
  * });
  *
- * @param {Object} option
```
Comment (fzaninotto) [inline packages/ra-core/src/controller/input/useReferenceArrayInputController.ts:31]: this change isn't necessary

File: packages/ra-ui-materialui/src/layout/AppBar.tsx [hunk 5]
```diff
@@ -59,13 +59,16 @@ const useStyles = makeStyles(
 /**
  * The AppBar component renders a custom MuiAppBar.
  *
- * @param {ReactNode} children React node/s to be render as children of the AppBar
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/layout/AppBar.tsx:62]: This change isn't necessary

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: packages/ra-core/src/controller/input/useReferenceArrayInputController.ts (hunk 4 above)
Comment (WiXSL) [reply packages/ra-core/src/controller/input/useReferenceArrayInputController.ts:31]: Are you sure? There is no param named `option` should be at least `* @param {Object} props`

File: packages/ra-ui-materialui/src/layout/AppBar.tsx (hunk 5 above)
Comment (WiXSL) [reply packages/ra-ui-materialui/src/layout/AppBar.tsx:62]: The same (I think), you have only one param named `props` here and no description for it

File: packages/ra-core/src/controller/input/useReferenceArrayInputController.ts (hunk 4 above)
Comment (WiXSL) [reply packages/ra-core/src/controller/input/useReferenceArrayInputController.ts:31]: Done!

File: packages/ra-ui-materialui/src/layout/AppBar.tsx (hunk 5 above)
Comment (WiXSL) [reply packages/ra-ui-materialui/src/layout/AppBar.tsx:62]: Done!

**Review by fzaninotto — APPROVED** [review]

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**